ordered_product/models.py

```diff
@@ -0,0 +1,101 @@
+                                 validators=[MinLengthValidator(6)],
+                                 null=False,
+                                 choices=[('client', 'client'), ('supplier', 'supplier')],
+                                 help_text="User Type - supplier/client")
+
+    class Meta:
+        abstract = True
+
+
+class Supplier(User):
+    business_name = models.CharField(max_length=32,
+                                     validators=[MinLengthValidator(6)],
+                                     null=False,
+                                     help_text="Business Name")
+
+
+class DeliveryLocation(models.Model):
+    user_name = models.ForeignKey(Supplier, on_delete=models.CASCADE)
+    location = models.CharField(max_length=32, validators=[MinLengthValidator(1)])
+    date = models.DateField()
+
+
+class Client(User):
+    area = models.CharField(max_length=32,
+                            validators=[MinLengthValidator(6)],
+                            help_text="area")
+
+
+class SupplierProduct(models.Model):
+    suppler_product_id = models.IntegerField(primary_key=True)
+    qr_code = models.ForeignKey(Product, on_delete=models.CASCADE)
+    user_name = models.ForeignKey(Supplier, on_delete=models.CASCADE)
+    price = models.IntegerField(validators=[MinValueValidator(1)])
+    quantity = models.IntegerField(validators=[MinValueValidator(1)])
+
+    class Meta:
+        unique_together = ('suppler_product_id', 'qr_code')
+
+
+class OrderdProduct(models.Model):
+    ordered_product_id = models.IntegerField(primary_key=True, editable=False, unique=True)
+    delivery_location_id = models.ForeignKey(DeliveryLocation, on_delete=models.CASCADE)
+    user_name = models.ForeignKey(Client, on_delete=models.CASCADE)
+    supplier_product_id = models.ForeignKey(SupplierProduct, on_delete=models.CASCADE)
+    quantity = models.IntegerField(default=1, validators=[MinValueValidator(1)])
+
+    @staticmethod
+    def filter_ordered_prodct_ID(opID):
+        return OrderdProduct.objects.filter(ordered_product_id=opID) if opID else None
+
+    @staticmethod
+    def filter_user_name(userName):
+        return OrderdProduct.objects.filter(user_name=userName) if userName else []
+
+    @staticmethod
+    def filter_delivery_location_id(dlID):
+        return OrderdProduct.objects.filter(delivery_location_id=dlID) if dlID else []
+
+    @staticmethod
+    def filter_supplier_product_id(spID):
+        return OrderdProduct.objects.filter(supplier_product_id=spID) if spID else []
+
+    @staticmethod
+    def get_all_ordered_product():
+        return OrderdProduct.objects.all()
+
+    def save_ordered_product(self):
+        self.save()
+        try:
+            OrderdProduct.full_clean(self)
+        except ValidationError as e:
+            self.delete()
+            raise e
+        return self
+
+    def delete_ordered_product(self):
+        OrderdProduct.objects.get(ordered_product_id=self.ordered_product_id).delete()
```

Comment:
You have increase,decrease,order(with quantity = 1) functions that needs to needs to be here and missing,Look at the DFD